optuna/storages/_rdb/storage.py

```diff
@@ -39,6 +41,30 @@
 _logger = optuna.logging.get_logger(__name__)
 
 
+@contextmanager
+def _scoped_session(scoped_session: orm.scoped_session) -> Generator[orm.Session, None, None]:
+    session = scoped_session()
+    try:
+        yield session
+        session.commit()
+    except IntegrityError as e:
+        _logger.debug(
+            "Ignoring {}. This happens due to a timing issue among threads/processes/nodes. "
+            "Another one might have committed a record with the same key(s).".format(repr(e))
+        )
+        session.rollback()
+        raise
```

Comment:
We're not "ignoring" the `IntegrityError` any longer here but instead propagating it. We have to handle this outside the caller of the `_scoped_session` with `try/catch` or introduce a parameter to this context manager whether it should be silenced or not. I personally prefer the former as the latter feel a bit ad-hoc. What do you think, are there perhaps better alternatives?

Reply:
I chose the latter one, because all algorithm related to session management can be defined by `_scoped_session`.
In my perspective, other people can understand the logic only checking the definition of `_scoped_session`.
https://github.com/optuna/optuna/pull/1628/commits/539d061df32554a29a22de63885605bb75d0df87

If I need to follow backward compatibility strictly, I need to ignore IntegrityError only for `self._commit_with_integrity_check`. What do you think of it?

Reply:
> If I need to follow backward compatibility strictly, I need to ignore IntegrityError only for self._commit_with_integrity_check. What do you think of it?

To follow it strictly, I committed it. https://github.com/optuna/optuna/pull/1628/commits/5166bec9c1f1a4f9ef2f2ded142fb084edd38165